### ISSUE TITLE
Can you let me do http again?

### ISSUE THREAD

**Issue by qtvr01** [seansfkelley/nas-download-manager#208]
As it was updated on the 22nd, http disappeared and only https were left. Can you let me do http again?

**Comment by patchper** [comment]
Same here. Failed to connect to NAS.

**Comment by 8bitgentleman** [comment]
Same here, can't connect to a non https address

**Comment by CyberMachin** [comment]
Same after PC reinstallation and configuring this Google Chrome extension.
Is this problem due to the DSM upgrade or the extension ?

**Comment by 8bitgentleman** [comment]
Personally I didn't have any upgrade between extension versions

**Comment by CyberMachin** [comment]
I try to do the same as my ancient configuration... I tried IP, NetBios, myDS nothing works. ;-(

**Comment by PatrickCras** [comment]
Same problem here, no connection after update.

**Comment by seansfkelley** [comment]
Unfortunately, no.

I had to make this change to comply with Firefox's [incoming add-on policies](https://extensionworkshop.com/documentation/publish/add-on-policies-dec-2021/) that are effective December 1st:

> Add-ons must use encryption when transporting data remotely.

As a workaround you can manually downgrade to 0.11.1, which presumably Firefox will continue to allow due to it not being the current version of the extension.

I'm sorry about the way this was rolled out -- I don't have any good messaging channels to warn about incoming changes, and I should have included a more-specific error message in the case where the extension knew you were using HTTP. That ship has sailed, so instead I've [added an entry to the FAQ](https://github.com/seansfkelley/nas-download-manager/blob/master/FAQ.md#why-cant-i-use-http-not-https-to-connect-to-the-nas) and I'll be pinning and updating this issue for anyone else who finds their way here.

**Comment by xBot55** [comment]
Same problem after uninstallation / installation for chrome and switch to HTTPS.

**Comment by seansfkelley** [comment]
@xBot55 that may be a proper bug. Can you take a look at https://github.com/seansfkelley/nas-download-manager/blob/master/FAQ.md and if that doesn't help, file a new issue with more detail?

**Comment by Thiefsie** [comment]
Is there an easy way to set up my synology to login with https?

Pity as I'm not running firefox but having the same problem

**Comment by razfag** [comment]
> Is there an easy way to set up my synology to login with https?
> 
> Pity as I'm not running firefox but having the same problem

I did run into the same problem but figured it out, the http download station api runs on port 5000 the https on 5001

- Goto https://<nas_ip_address>:5001 in your browser
- In the invalid certificate warning advanced -> confirm security exception
- change the port in the addon settings to 5001

**Comment by Thiefsie** [comment]
> > Is there an easy way to set up my synology to login with https?
> > Pity as I'm not running firefox but having the same problem
> 
> I did run into the same problem but figured it out, the http download station api runs on port 5000 the https on 5001
> 
> * Goto https://<nas_ip_address>:5001 in your browser
> * In the invalid certificate warning advanced -> confirm security exception
> * change the port in the addon settings to 5001

Thanks you very much. That worked a treat.

**Comment by qtvr01** [comment]
> > > Is there an easy way to set up my synology to login with https?
> > > Pity as I'm not running firefox but having the same problem
> > 
> > 
> > I did run into the same problem but figured it out, the http download station api runs on port 5000 the https on 5001
> > 
> > * Goto https://<nas_ip_address>:5001 in your browser
> > * In the invalid certificate warning advanced -> confirm security exception
> > * change the port in the addon settings to 5001
> 
> Thanks you very much. That worked a treat.

In the invalid certificate warning advanced -> confirm security exception

Can you tell me in detail where to set this up?

**Comment by qtvr01** [comment]
> > Is there an easy way to set up my synology to login with https?
> > Pity as I'm not running firefox but having the same problem
> 
> I did run into the same problem but figured it out, the http download station api runs on port 5000 the https on 5001
> 
> * Goto https://<nas_ip_address>:5001 in your browser
> * In the invalid certificate warning advanced -> confirm security exception
> * change the port in the addon settings to 5001

In the invalid certificate warning advanced -> confirm security exception

Can you tell me in detail where to set this up?

**Comment by razfag** [comment]
@qtvr01 Not sure what to tell you. It's called differently on chome & firefox but behaves the same.

Eventually  you have to login on the nas with the same credentials on the nas afterwards inside that browser tab over https once.

**Comment by borse2008** [comment]
This guide helped me configure DSM 7 from http setup to https.  Now working for me.

https://mariushosting.com/synology-how-to-enable-https-on-dsm-7/

**Comment by Shawzborne** [comment]
So I managed to get DSM 6.2.4 switched over to HTTPS on my DS1817+ however I would appreciate if someone can post an easier guide than this one [](https://kb.synology.com/en-my/DSMUC/tutorial/How_to_enable_HTTPS_and_create_a_certificate_signing_request_on_your_Synology_Unified_Controller on how to setup a certificate for HTTPS/SSL) so that we don't get the certificate warnings every time DSM is opened in chrome. Please and thank-you

**Comment by papinist** [comment]
> > Is there an easy way to set up my synology to login with https?
> > Pity as I'm not running firefox but having the same problem
> 
> I did run into the same problem but figured it out, the http download station api runs on port 5000 the https on 5001
> 
> * Goto https://<nas_ip_address>:5001 in your browser
> * In the invalid certificate warning advanced -> confirm security exception
> * change the port in the addon settings to 5001

This should be pinned on top.. thank you!

**Comment by Otto42** [comment]
Still doesn't work. How does one manually downgrade? Where is the old 0.11.1 version available from?

**Comment by wickedgalen** [comment]
Hi,

I got the same issue.

Experience so far:

**Windows 10** : the solution was installing, self-signed certificates, generated by my NAS. This worked more or less smooth, although the issue will come back as the certificate will only last 1 year. (.cer files)

**Linux** : although I installed the same certificates (conversion from .pem to .crt , update & reboot ) the Browser does not accept it I cannot setup the Add-On. (The process here is quite painful compared to Windows)

Additionally for both systems or any :

Could the add-on at least save the settings although not working ? avoiding to retype them every time ?

Also I specifically configured my NAS to send all HTTP traffic to HTTPS in case of missing the "S" resolving in a Forbidden Error, although this could be a NAS issue.

Also I noticed Chrome seems to have less difficulties accepting the certificates than Firefox.

The only thing working is directly accessing the NAS sadly defying the purpose of the addon.

Simply accepting the security risk in the browser is rarely working.

I hope this can be fixed.

If someone has another working solution I'm eager to try.
